src/test/ui/type-alias-impl-trait/bound_reduction2.stderr

```diff
@@ -1,4 +1,4 @@
-error: defining opaque type use does not fully define opaque type
+error: defining opaque type use does not fully define opaque type: generic parameter `V` is specified as concrete type `<T as TraitWithAssoc>::Assoc`
```

Comment:
We could improve this error. (Also, I think `define` should be `determine`.) Maybe with a help message like:
```
uses of an opaque type must fully determine it, but here `Foo` is only determined for `V = <T as TraitWithAssoc>::Assoc`
```
It took me a couple of tries to work out what the error message was saying before.

Reply:
@varkor: Your proposed error message is backward - `V` is the parameter that is *not* determined (because it's a concrete type instead of a generic parameter).

Reply:
While I think there's room to improve this error, this modification provides strictly more information. I think further improvements should go in another PR, since this PR is really about changing *when* error messages occur, not the actual contente.

Reply:
Okay, but let's open an issue for improving this.

Reply:
Opened https://github.com/rust-lang/rust/issues/66723.